src/main/java/org/broadinstitute/hellbender/tools/copynumber/CollectPerBaseCounts.java

```diff
@@ -0,0 +1,170 @@
+package org.broadinstitute.hellbender.tools.copynumber;
+
+import com.google.common.collect.ImmutableList;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.CoverageAnalysisProgramGroup;
+import org.broadinstitute.hellbender.engine.AlignmentContext;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.LocusWalker;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.filters.MappingQualityReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.tools.copynumber.arguments.CopyNumberArgumentValidationUtils;
+import org.broadinstitute.hellbender.tools.copynumber.datacollection.PerBaseCountCollector;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.PerBaseCountCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.metadata.Metadata;
+import org.broadinstitute.hellbender.tools.copynumber.formats.metadata.MetadataUtils;
+import org.broadinstitute.hellbender.tools.copynumber.formats.metadata.SampleLocatableMetadata;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Collects per-base counts at specified sites where bases are one of those specified in {@link PerBaseCountCollector#BASES}.
+ * If two reads share the same query name and, therefore, originate from the same template of DNA, this
```

Comment:
The BAM snippets were indeed chosen because they contain paired-end reads that overlap at some sites of interest. This process in general is more of an issue in cell-free DNA where the vast majority of templates are ~166bp long, which is shorter than twice the read length. The result is that many bases on a given template are reported twice, once from each paired-end read. As far as exact numbers of how often this happens, it is a function of read length and the distribution of template lengths.

Reply:
OK, thanks @rklein2, it might be useful to note that somewhere.

Now I'm wondering how effective it might be to just use FirstOfPairReadFilter in CollectAllelicCounts in typical data (I haven't looked at much cfDNA data closely, though)?  I believe that most of the sites in our common list are farther apart than ~166bp, and I wouldn't expect that differing BQs would decide calls that often.  You might lose some depth in the higher peaks/tail of the distribution, but perhaps this might be a stopgap we can recommend until the tools are consolidated.